src/mlpack/core/data/string_algorithms.hpp

```diff
@@ -0,0 +1,50 @@
+/** 
+ * @file core/data/string_algorithms.hpp
+ * @author Gopi M. Tatiraju
+ *
+ * Utility fucntions related to string manipulation
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_STRING_ALGORITHMS_HPP
+#define MLPACK_CORE_DATA_STRING_ALGORITHMS_HPP
+
+namespace mlpack{
+namespace data{
+
+/**
+ * A simple trim fucntion to strip off whitespaces 
+ * from both the side of string.
+ */
+inline std::string trim(std::string str)
```

Comment:
Do not hesitate in adding a Doxygen compatible documentation for this function :+1: